### ISSUE TITLE
xEdit AKA FO4Edit, TES5Edit integration

### ISSUE THREAD

**Issue by Lii-Kun** [sonic2kk/steamtinkerlaunch#700]
## System Information
- SteamTinkerLaunch version: 12
- Distribution: POP!_OS

## Feature Description

I used xEdit a lot to mod Bethesda games when I was running windows but haven't found a way to get it running correctly in Linux. The different versions are the same .exe just renamed. So it just needs 1 install for all the games it supports. So please put this in or point me to a way to get it working

I guess I should put a quick "what is xEdit" Its a tool that can be used to make or modify plugin files for Bethesda games such as Fallout 4, Fallout 3, Fallout New Vegas, Skyrim, Morrowind, Oblivion. It can also be used to "clean" plugin files.

**Comment by sonic2kk** [comment]
Hi!

I wouldn't strictly be _opposed_ to adding support for these tools, but you mentioned that you can't get them to work correctly under Linux. SteamTinkerLaunch can't guarantee support for these tools either I'm afraid, so if it doesn't work when running manually, it probably won't work with SteamTinkerLaunch either. You'll need to find a way of getting them to work on Linux before STL could support them.

> or point me to a way to get it working

I do not and have not ever used these tools, but MO2 lets you add custom executables and in the past some tools I have used seemed to work. I would guess though that if these tools don't work on Linux in general that they won't work here either.

SteamTinkerLaunch's Windows program stuff just runs various programs using Wine/Proton, and where necessary performs known tweaks to improve compatibility - Note: *known* tweaks. There's no extra magic there, so if these tools just flat-out don't work on Linux and there are no known fixes, there isn't much we can do either I'm afraid.

Putting that aside, I don't know or use these tools so adding support may be better left to someone that knows the ins and outs much better here. There are other considerations too, such as are we allowed to automatically download these tools, what paths do they need to be put in, and so on. All of this on top of the question of compatibility.

<hr>

So in short, this is probably not something that *I* will do, but it's also something I'd happily review a PR for. I'll leave this issue open for a bit for discussion on how to get these tools running in general and at the very least a section on the [modding wiki](https://github.com/sonic2kk/steamtinkerlaunch/wiki/Modding) can be added to describe how to get these tools to work.

Thanks!

**Comment by Lii-Kun** [comment]
Some people say they have it working. [Here.](https://www.afkmods.com/index.php?/topic/5509-xedit-and-linux-tutorial/) I'm a linux noob only been doing this 2 weeks or so. I launched it with bottles to see what would happen. xEdit looks at windows registry to see where the game is installed and obviously couldn't find the game. And it checks the my documents folder for the games .ini file. I think xEdit needs to be in the same wine prefex as the game but not sure how to do that.

**Comment by sonic2kk** [comment]
That guide is a bit old but looks fairly straightforward, thanks for linking it! The main "extra" steps seem to be creating a symlink for the game INI file in the xEdit downloads folder, but I am not sure if that would be strictly required if xEdit is running in the same prefix as the game as you pointed out.

The impression I am getting is that xEdit is just a standalone executable. If that is correct, you may be able to run your desired xEdit exeuctable using SteamTinkerLaunch's one-time run. Launch the game you want to run xEdit with (e.g., Fallout New Vegas) and on the SteamTinkerLaunch main menu, click the "One-Time Run" button. From here you'll see the One-Time Run dialog

![image](https://user-images.githubusercontent.com/7917345/209994504-6ec1045d-cbf2-4872-809e-058c0b306ce5.png)

For "one time command" navigate to the xEdit executable. You *may* need to check "Use custom executable directory as working directory" if the xEdit executable relies on various files in that working directory, as some tools do require this. Once you're ready you can click "Run command" and that will run the xEdit executable with the given version of Proton in the current game's prefix.

I am not totally sure if this will work but with my *very* surface level knowledge from skimming that guide, this may work. It should be able to see the registry keys for the installed game at least. Give it a try and see how you get on :-) I don't think using MO2 would work to launch it either, since as you mentioned it requires the same prefix as the game and I think even after the overhaul, MO2 runs in its own prefix with STL. One-Time Run however should use the game's prefix. There could be other issues there such as the working directory but the symlink step could be performed manually.

<hr>

Also, is xEdit available anywhere other than Nexusmods? For example, perhaps on GitHub? If not, it may not be possible to provide integration with it, as we can't download programs from Nexus.

**Comment by Lii-Kun** [comment]
I got it working just now by adding it to vortex. I'm guessing it will also work with mo2 the same way

**Comment by sonic2kk** [comment]
No worries, glad it works! It's probably more convenient to run it through the mod manager too.

If it's just as straightforward as adding it to the mod manager, I don't think any further integration will be needed on the STL side.

Closing this issue now, thanks for the discussion! I will add a note about this on the modding wiki just in case other users are curious :-)
